Cataclysm/Druid.lua

```diff
@@ -236,37 +262,42 @@ spec:RegisterStateFunction("set_last_finisher_cp", function(val)
     lastfinishercp = val
 end)
 
-local riptfsnapshot = nil
-spec:RegisterStateExpr("rip_tf_snapshot", function()
-    return riptfsnapshot
+spec:RegisterStateExpr("pseudo_rip_tf_snapshot", function()
+    if tracked_rips[ target.unit ] then
+       return tracked_rips[ target.unit ].tf_snapshot 
+    end
+    return "leer"
 end)
 
-spec:RegisterStateFunction("set_rip_tf_snapshot", function(val)
-    riptfsnapshot = val
+spec:RegisterStateExpr("rip_tf_snapshot", function()
+    return rip_tracker[target.unit].tf_snapshot
 end)
 
-local rip_extend_counter = 0
-spec:RegisterStateExpr("rip_extend_count", function()
-    return rip_extend_counter
-end)
+local ExpirePrimalMadness = setfenv( function ()
 
-spec:RegisterStateFunction("set_rip_extend_count", function(val)
-    rip_extend_counter = val
-end)
+    if buff.primal_madness.up then
```

Comment:
We probably want to add some sanity checking here since it seems TF and Berserk can both proc Primal Madness.  Do we need to check, at these anticipated expiration times, whether TF and Berserk are both down before removing the Energy?  Otherwise, a `cancel_buff` recommendation just before a PM expires could pretend to remove the energy twice.